AEPMessaging/Tests/UnitTests/schemas/FeedItemSchemaDataTests.swift

```diff
@@ -15,11 +15,194 @@ import XCTest
+
+        // test
+        guard let encodedFeedItem = try? encoder.encode(feedItem) else {
+            XCTFail("unable to encode FeedItemSchemaData")
+            return
+        }
+
+        // verify
+        let actual = getAnyCodable(String(data: encodedFeedItem, encoding: .utf8) ?? "")
+        assertExactMatch(expected: expected, actual: actual)
     }
     
+    func testIsEncodableStringContent() throws {
+        // setup
+        let feedJson = "{\"expiryDate\":\(mockExpiry),\"meta\":{\"\(mockMetaKey)\":\"\(mockMetaValue)\"},\"content\":\"\(mockContentValue)\",\"contentType\":\"\(ContentType.textPlain.toString())\",\"publishedDate\":\(mockPublished)}"
+        guard let feedItem = getDecodedFeedItem(fromString: feedJson) else {
+            XCTFail("unable to decode feedJson")
+            return
+        }
+        let encoder = JSONEncoder()
+        let expected = getAnyCodable("{\"expiryDate\":\(mockExpiry),\"meta\":{\"\(mockMetaKey)\":\"\(mockMetaValue)\"},\"content\":\"\(mockContentValue)\",\"contentType\":\"\(ContentType.textPlain.toString())\",\"publishedDate\":\(mockPublished)}") ?? "fail"
+
+        // test
+        guard let encodedFeedItem = try? encoder.encode(feedItem) else {
+            XCTFail("unable to encode FeedItemSchemaData")
+            return
+        }
+
+        // verify
+        let actual = getAnyCodable(String(data: encodedFeedItem, encoding: .utf8) ?? "")
+        assertExactMatch(expected: expected, actual: actual)
+    }
+    
+    // Exception paths
+    func testContentIsRequired() throws {
+        // setup
+        let feedJson = "{\"expiryDate\":\(mockExpiry),\"meta\":{\"\(mockMetaKey)\":\"\(mockMetaValue)\"},\"contentType\":\"\(mockContentType.toString())\",\"publishedDate\":\(mockPublished)}"
+        
+        
+        // test
+        let feedItem = getDecodedFeedItem(fromString: feedJson)
+
+        // verify
+        XCTAssertNil(feedItem)
+    }
+    
+    func testContentTypeIsRequired() throws {
+        // setup
+        let feedJson = "{\"expiryDate\":\(mockExpiry),\"meta\":{\"\(mockMetaKey)\":\"\(mockMetaValue)\"},\"content\":{\"\(mockContentKey)\":\"\(mockContentValue)\"},\"publishedDate\":\(mockPublished)}"
+        
+        
+        // test
+        let feedItem = getDecodedFeedItem(fromString: feedJson)
+
+        // verify
+        XCTAssertNil(feedItem)
+    }
+    
+    func testPublishedDateExpiryDateMetaAreOptional() throws {
+        // setup
+        let feedJson = "{\"content\":{\"\(mockContentKey)\":\"\(mockContentValue)\"},\"contentType\":\"\(mockContentType.toString())\"}"
+        
+        
+        // test
+        let feedItem = getDecodedFeedItem(fromString: feedJson)
+
+        // verify
+        XCTAssertNotNil(feedItem)
```

Comment:
nit: Can you please verify the content and content type here since feed item is not nil?

Reply:
done